rust/core-lib/src/file.rs

```diff
@@ -136,12 +178,23 @@ impl FileManager {
     fn save_new(&mut self, path: &Path, text: &Rope, id: BufferId) -> Result<(), FileError> {
         try_save(path, text, CharacterEncoding::Utf8)
             .map_err(|e| FileError::Io(e, path.to_owned()))?;
+
+        #[cfg(feature = "notify")]
+        let new_tail_details = TailDetails {
+            current_position_in_tail: 0,
+            is_tail_mode_on: false,
+            is_at_bottom_of_file: false,
```

Comment:
this might also be something we can determine from the state of the view; if `view.first_line + view.height` == the total number of lines, we know the last line is visible.

Reply:
This is much better than current implementation. Thanks for suggestion. :)

Reply:
thinking about this more, getting the total number of visual lines in the document might be slightly complicated right now; if you get stuck let me know.